### DISCUSSION TITLE
FDS Input File Guidance

### DISCUSSION THREAD

**Discussion by mjerebets** [firemodels/fds#13526] [Q&A] [answered]
Hi, 
I have a few questions about my set-up in FDS, and want to make sure that my guiding assumptions are correct. I’m working on a set of cases where I’m looking at fire behavior as a function of ignition pattern, with varying wind speeds and ignition lengths. I would really appreciate some help clarifying the following: 

1. For my vegetative fuel, I followed the inputs in the CSIRO grassland file, using the particle fuel method (LPM). I understand that the grid size chosen affects the smallest computational unit- **so is it correct to think that my fuel height must be a multiple of this grid size?** Ex: &INIT PART_ID = ‘blade of grass’, XB = -10.0, 40.0, -20.0, 20.0, 0.0, 0.5. This is not the case in the CSIRO input files (specifically C064_fine), where particle height is 0.21 m. I would appreciate some clarification about the mechanisms here. 
2. Along that vein, **is the extinction length of particles something I should be concerned about** (re: [Morvan 2011](https://www.researchgate.net/profile/Dominique-Morvan/publication/225661352_Physical_Phenomena_and_Length_Scales_Governing_the_Behaviour_of_Wildfires_A_Case_for_Physical_Modelling/links/00b7d51b88039f3785000000/Physical-Phenomena-and-Length-Scales-Governing-the-Behaviour-of-Wildfires-A-Case-for-Physical-Modelling.pdf), [Mueller et al., 2020](https://research.fs.usda.gov/treesearch/60868))? General guidance is that the grid size should be 3x smaller than the extinction length, so that would lead me to a grid cell size of 5cm (very computationally expensive for a 100,000 m² domain). However, it seems like this was not taken into account for the CSIRO input files, which have a cell size of 25 cm. 
3. I have seen some discussion about where SLCF and devices should be located. Is there a potential problem with placing these slices and devices at the edge of grid cells as in my input file, or is this the proper placement (with CELL_CENTERED = F)? In other words, say 0.5m cell size and a line of temperature devices at x = 0 or x = 0.5 m. **Is it correct thinking that the values will be taken from the vertices of the cell? Does this work by averaging the 4 cell values incident to the vertex?** 
4. The way I set up wind is similar to the CSIRO inputs, with one main change- I wanted to initialize the wind vector immediately using U0. **Will this then become the M-O wind profile over time?** &WIND_SPEED = 1.0, DIRECTION = 270., L = -500., Z_0 = 0.03, U0 = 1.0, V0 = 0, W0 = 0. Does this seem reasonable, or should I wait to begin ignition until after my wind field has developed, rather than specifying U0? And, what is the z_0 representing? 
5. My next step before running the suite of simulations with different winds/ignition lengths is to do a domain convergence test and a grid cell size convergence test. I currently have a buffer of at least 10m in the x- and y-direction, and 45m in the z-direction, where no ignition is taking place. It doesn’t seem (at least in Smokeview) that any edge effects are taking place. **Is there a good rule of thumb for where I should start with my domain convergence test? Should I go much bigger/smaller?** I wasn’t able to find any guidance for this, so I would really appreciate any insight for a good starting point. 
6. Relatedly, I have seen the guidance around a plume resolution index (D*/𝛿x) to inform a grid convergence test. With my maximum HRR, it seems like the grid sizes I should test are 0.5m, 0.2m, and 0.125m. **Does this seem like a reasonable method to begin with, or would you suggest something different?** 

I would really appreciate any feedback about my methods! Attached is my most simple input file, which leverages the vegetation_model.txt file found in the CSIRO cases. My ignition pattern is a V-shape at a 90 degree angle. And please let me know if there is any more information I can provide. (I have removed all my outputs of interest for the sake of brevity in the FDS file). 

[simple_chevron_case.txt](https://github.com/user-attachments/files/17264377/simple_chevron_case.txt)

Thank you,
Marta

**Comment by ericvmueller** [comment, marked as answer]
Hi Marta, here are some thoughts about these points:

1. Using the typical approach of N_PARTICLES_PER_CELL = 1, there will be a particle placed in every cell that intersects with the XB volume. If some of these cells have only part of their volume filled by fuel – say because fuel height is not a multiple grid cell size – then the bulk density represented by the Lagrangian particle will be rescaled in those cells. The basic criteria is to have the correct total mass of fuel based on the user specified XB and MASS_PER_VOLUME. The effect this might have on your case is something you need to consider in your sensitivity analysis.

2. Yes this is an important length scale but its importance is also related to the overall role of radiation in your flame spread scenario (e.g compared to convection). Again, this is still an area of research for the model and so it is worth exploring the sensitivity. You might consider stretched grids to get higher resolution in the fuel layer without tanking your runtime.

3. In general, devices take the nearest value without interpolation (e.g. cell center for temperature, cell face for velocity components). Slices interpolate to nodes unless using CELL_CENTERED = T. 

4. Using the SPEED, DIRECTION, etc. parameters should already initialize the wind profile. It wont spin up from zero, so you don’t need to use U0,V0,W0. 

5. It is always best to keep the boundaries away from the region of interest. Some improvements have been made with the most recent release to improve stability and reduce upwind influence of open wind boundaries …but there can still be influences of things like having a steady inflow wind transition to turbulence. Having at least 10m is probably a good starting point if your fire line depth is on the order of 10m or less but I don’t have a great rule of thumb other than testing the influence.

6. Depending how you define it, I think this use of D_star is more appropriate for a stationary axisymmetric plume. For a line fire, if D_star gets larger the longer the fireline it might suggest you need less resolution when the flame spread locally will still be defined by things like the flame length and fuel depth which will not depend on the overall fireline length.
Maybe think in terms of fuel depth, which your suggested resolutions might be multiples of. 

A lot of your points have to do with convergence, and it is not straightforward for these complex flame spread cases. The best advice I can give is to explore the sensitivity as much as you can for this kind of research.

**Comment by mjerebets** [comment]
Hi Eric, 

Thank you for taking the time to answer all of my questions, I really appreciate it. It sounds like I have some extra sensitivity analyses to do before running my set of simulations, but it's good to know that I'm not entirely off-base with these concepts. 

Thanks again,
Marta
